In [199]:
import pandas as pd
import re

In [200]:


import numpy as np
from collections import Counter
def post(x,topk,frc):
    #格式处理：c
#     x = np.array(x).flatten().tolist()
#     x=' '.join(x)
    x = x.split() 
    #过滤
#     r = []
#     for i in Counter(x).most_common(10000):
#         if i[1]>1:
#         #print(i)
#         r.append(i)
#     return r
    return[i[0] for i in  Counter(x).most_common(topk) if i[1]>frc and len(i[0])>1]


def make_df_entity_1(path,flag='train'):
    out_path = path+'.df_entity_1.csv'
    df1 = pd.read_csv(path,sep='\t')
    df1['top_esm_entity']=df1.esm_content.map(lambda x:post(x,topk=30,frc=2))
    df1['top_esm_title']=df1.esm_title.map(lambda x:post(x,topk=5,frc=0))



    newlines=[]
    for i,line in df1.iterrows():


        texta =line['content']
        
        if flag=='test':
            entity_real=['null']
        else:
            entity_real = line['entity'].split()
        entity_recall = line['top_esm_title'] + line['top_esm_entity']
        for entity in entity_recall:
            textb = entity
            if entity in entity_real:
                label = 1
            else:
                label = 0
            newlines.append([line['newsId'],texta,textb,label])
    df_entity = pd.DataFrame(newlines,columns=['newsId','texta','textb','label'])
    df_entity.to_csv(out_path,index=False,sep='\t')




In [276]:
def make_df_entity_2(path,flag='train'):
    agg = True
    if agg:
        out_path = path+'.agg.df_entity_2.csv'
    else:
        out_path = path+'.df_entity_2.csv'
    print(out_path)
    df1 = pd.read_csv(path,sep='\t')
    df1['top_esm_entity']=df1.esm_content.map(lambda x:post(x,topk=30,frc=2))
    df1['top_esm_title']=df1.esm_title.map(lambda x:post(x,topk=5,frc=0))

    newlines=[]
    for i,line in df1.iterrows():

        entity_recall = line['top_esm_title'] + line['top_esm_entity']
        sentences = re.split(r'。|？|！', line['content'])
        if flag =='test':
            entity_real=['null']
        else:
            entity_real = line['entity'].split()
        
        entity_real = [clean_texts(x) for x in entity_real]
        for sentence in sentences:
            sentence = clean_texts(sentence)
            for j ,entity in enumerate(entity_recall):

                if entity[0] in ['*','+']:
                    entity='\\'+entity
                if  entity=='c++':
                    entity='c\++'

                textb = entity
                if entity in entity_real:
                    label = 1
                else:
                    label = 0
                try:
                    if re.search(entity,sentence):
                        newlines.append([line['newsId'],sentence,entity,label])
                except:
                    pass
                    #print(entity,sentence)


    df_entity = pd.DataFrame(newlines,columns=['newsId','texta','textb','label'])
    #df_entity.dropna(inplace=True)
    
    if agg:
        gg = df_entity.groupby(['newsId','textb'])
        df_agg2 = gg['texta'].apply(lambda x:','.join(x)).reset_index()
        df_agg2['label'] = gg['label'].apply(list).reset_index()['label']
        df_agg2['label'] = df_agg2['label'].map(lambda x:x[0])

        df_entity=df_agg2



    df_entity.dropna(inplace=True)

    print(df_entity.head())
    df_entity.to_csv(out_path,index=False,sep='\t')




In [277]:
import re
def clean_texts(x):
        
        #return  re.sub("[+——()?（）\\n \\t ]+", "",x)
        return re.sub("[\s+\.\!\/_,$%^*(+\"\')]+|[+——()?【】“”！，。？、~@#￥%……&*（）]+", "",x)


In [278]:
path = '/sdb1/zhangle/2019/zhangle11/sohu-2019/data/coreEntityEmotion_train.txt.esm_entity.csv'
#make_df_entity_1(path)
make_df_entity_2(path)

path2 = '/sdb1/zhangle/2019/zhangle11/sohu-2019/data/coreEntityEmotion_test_stage1.txt.esm_entity.csv'
#make_df_entity_1(path2,'test')
make_df_entity_2(path2,'test')


/sdb1/zhangle/2019/zhangle11/sohu-2019/data/coreEntityEmotion_train.txt.esm_entity.csv.agg.df_entity_2.csv
     newsId textb                                              texta  label
0  0000214b    12  朋友圈大神解读如下现在假设你的采购成本是100元成交价是120元,如果按17增值税率你的进项...      0
1  0000214b    16  蓝字文章来源：服装邦3月15日两会答记者问总理透露今年下决心要进行更大规模的减税降费4月1日...      0
2  0000214b    价格  我们抛开企业竞争力量之说仅就降税了对企业的价格是怎么影响的来给提出降价要求的企业和应对降价要...      0
3  0000214b    企业  这本是对企业的一项重大利好,这个客户说降一个点还可以接受现在是降三个点客户是上市公司强势去年...      0
4  0000214b    减税  蓝字文章来源：服装邦3月15日两会答记者问总理透露今年下决心要进行更大规模的减税降费4月1日...      0
/sdb1/zhangle/2019/zhangle11/sohu-2019/data/coreEntityEmotion_test_stage1.txt.esm_entity.csv.agg.df_entity_2.csv
     newsId textb                                              texta  label
0  0000327e    价格  一国内焦亚硫酸钠价格走势图据生意社监测本周国内焦亚硫酸钠稳定前行工业级焦亚硫酸钠周初均价19...      0
1  0000327e    企业  焦亚硫酸钠行业年后生产较为稳定企业整体库存尚可多以完成老客户订单为主新单增加有限交投稍显清淡...      0
2  0000327e    市场  二行情分析产品：本周焦亚硫酸钠市场整体行情相对稳定前行工业级焦亚硫酸钠市场价格区间1850-...      0
3  0000327e    整体  二

In [286]:
p = '/sdb1/zhangle/2019/zhangle11/sohu-2019/data/coreEntityEmotion_train.txt.esm_entity.csv'
df=pd.read_csv(p,sep='\t')

In [288]:
df.content.map(lambda x:len(x)).mean()

1293.505175

In [283]:
df.shape

(686369, 4)

In [284]:
df.head()

newsId textb                                              texta  label
0  0000214b    12  朋友圈大神解读如下现在假设你的采购成本是100元成交价是120元,如果按17增值税率你的进项...      0
1  0000214b    16  蓝字文章来源：服装邦3月15日两会答记者问总理透露今年下决心要进行更大规模的减税降费4月1日...      0
2  0000214b    价格  我们抛开企业竞争力量之说仅就降税了对企业的价格是怎么影响的来给提出降价要求的企业和应对降价要...      0
3  0000214b    企业  这本是对企业的一项重大利好,这个客户说降一个点还可以接受现在是降三个点客户是上市公司强势去年...      0
4  0000214b    减税  蓝字文章来源：服装邦3月15日两会答记者问总理透露今年下决心要进行更大规模的减税降费4月1日...      0

In [194]:
p = '/sdb1/zhangle/2019/zhangle11/sohu-2019/data/coreEntityEmotion_train.txt.esm_entity.csv.df_entity_1.csv'
df=pd.read_csv(p,sep='\t',nrows=100)

In [195]:
df.head()

newsId                                              texta textb  label
0  4e36d02a  第十七届上海国际工业自动化及机器人展与上海智能工厂展览会于2019年3月1日圆满落下帷幕。展...   sia      0
1  4e36d02a  第十七届上海国际工业自动化及机器人展与上海智能工厂展览会于2019年3月1日圆满落下帷幕。展...    智慧      0
2  4e36d02a  第十七届上海国际工业自动化及机器人展与上海智能工厂展览会于2019年3月1日圆满落下帷幕。展...  智慧工厂      0
3  4e36d02a  第十七届上海国际工业自动化及机器人展与上海智能工厂展览会于2019年3月1日圆满落下帷幕。展...    机器      0
4  4e36d02a  第十七届上海国际工业自动化及机器人展与上海智能工厂展览会于2019年3月1日圆满落下帷幕。展...    智能      0

In [37]:
path = '/sdb1/zhangle/2019/zhangle11/sohu-2019/data/coreEntityEmotion_train.txt.esm_entity.csv'

In [38]:
df1 = pd.read_csv(path,sep='\t')

In [39]:
df1.head(100)

newsId                                            content  \
0   4e36d02a  第十七届上海国际工业自动化及机器人展与上海智能工厂展览会于2019年3月1日圆满落下帷幕。展...   
1   cb8e8b79  大数据之热度，已无需多言。业内众多关于大数据可视化应用领域的声音与讨论，大多集中在行业领域和...   
2   f85c18e2  过年期间和一些搞运输的老朋友聊天，大家始终绕不开的一个话题：“如今物流行业越来越难干了！”想...   
3   889199b4  水泥路面起砂如何处理？修补后材料会发生脱落吗？水泥路面起砂病害的发生可能是材料老化的现象，也...   
4   a811d20e  我们今天再开始复盘之前，先简单说下行情，昨天大幅度波动90点的时候，很多人看空，但是却没有勇...   
5   4e798999  对于大部分入门玩家来说，在电源的选择上往往都比较随意。他们在意的方面一般只有功率的大小，而对...   
6   1f5679fb  新打的水泥地面受冻掉皮怎么修复？水泥路面、地面掉皮病害发生后，有一些水泥路面、地面发生病害后...   
7   1a8339ea  一、信托融资概念及特点\n信托是以资产为核心，信任为基础，委托为方式的财产管理制度。信托融资...   
8   51203867  当“租购并举”成为国策，住房租赁的庞大市场需求和政策红利的持续加持，让越来越多人发现并逐步重...   
9   5bb9047a  随着生活压力和工作压力都不断增加的情况下，有心理问题的人也越来越多，很多人都开始求助心理咨询...   
10  478af383  标普500成分股公司在去年第四季度关心什么？摩根大通从它们的电话会议记录、问答环节记录以及成...   
11  f0a7d8d9  随着移动支付的便民化与普及，现在基本上在三线城市以上，不论走到何地，手机一扫，分分钟就能付款...   
12  afee8716  新倚天屠龙记：灭绝硬撑最后一口气，爆出惊人秘密，让她崩溃大哭\n大型古装武侠剧《新倚天屠龙记...   
13  efd34fe8  中字头国企整体产权转让。公司名下核心资产为廊坊市区三块工业用地，合计面积约220亩。公司自有...   
14  73ce42e1  玩具行业一周简讯，宜采网整理报道：\n【简讯一】\n喜讯！制造业税率降3个点，电价降10%！...   
15  783adc02  江苏天鼎看股票：盘中盯花眼，不如盘前下功夫！\n\n在前文中江苏天鼎简单为大家介绍了集合竞价...   
16  960b8e04  2018年债牛行情一骑绝尘，2019年a股“牛市”横空演绎，债市耀眼的光环就这样被a股取代了...   
17  d7cb8b85  \n为什么下跌的黄金走到1282至1280一线之后，却停止不前了，到底是见底了还是在酝酿反弹...   
18  66fb9c0c  导读：蔚来汽车三年累计亏损已达172.3亿。不过，从积极的一面来看，随着蔚来的第二款量产车型...   
19  cd510c0d  \n胡润研究院携手小小世界全球魅力女性商学院3月7日联合发布《2019胡润全球白手起家女富豪...   
20  80800a17  今天市场不敢发表太多，但是大家留意两会后，还有近期会有大幅度的波动，好似蹦迪那样的感觉。如果...   
21  2322249a  实验室建设不同于普通的建设项目，具有综合性、专业性、系统性强的特点，必须统一部署，总体协调，...   
22  ac148da5  印度标准局（bis）是印度标准及认证事务的主管部门。\n\n历史沿革\n\n印度在独立前没有...   
23  fe0ae621  上篇文章我们来了解了亚马逊跨境电商的优点，便捷链接：花无百日红，唯你一支独秀，亚马逊跨境电商...   
24  530ccf21  商场触摸查询广告机是交互3d高科技以及互联网科技来和谐商业连锁店与顾客间在出售、营销和服务上...   
25  7885555d  ​​百度竞价推广是一种按效果收费的广告，是目前互联网行业较为精准的广告投放方法之一，而随着愈...   
26  c494a2ab  虽然科技发展，社会进步，但很多人特别是老一辈人在子女的婚恋方面还是多数会选择相亲的方式。而与...   
27  779320d5  2019年1月25日，重庆水利电力职业技术学院水利工程学院副院长闵志华，水工造价专任教师雷丽...   
28  46f7801f  三月阳光正好，三月微风不燥，\n春暖花开，柳暖花香，正如你如此美好\n确认过眼神，遇见最美丽...   
29  ed3a5d29  \n作为企业管理者，担当是必备的素质之一，你懂得如何担当吗？而“担当”的内涵包括：提升能力，...   
..       ...                                                ...   
70  5f2de988  涵艺投资前言：\n每一天市场都充斥着无数的消息与文章，每一天都会有盈利百万的神单刺激着我们的...   
71  254203ca  想必大家对2008年的金融危机都还有印象吧？这场危机源于美国，波及全世界，连俺们村的小厂子都...   
72  b8caaa52  商标作为企业和个人的形象代表，而版权分为软件著作权和作品著作权，但仍有很多企业对于两者之间的...   
73  e3c6a6f4  父母对孩子的期望有很多，但是孩子自己的梦想却非常趋同，每个女孩都有一个“公主梦”，每个男孩则...   
74  317494c0  随着昨天上证指数突破3100点。仿佛一夜之间，各种牛市来了的声音就铺天盖地席卷而来了。回头想...   
75  bd7f602c  \n在楼市，去年开始出现一个新词，叫做“降价未遂”。近期，又有降价未遂事件发生，据中国江西网...   
76  922c203c  “如中大布匹市场搬迁，加上康乐村改建高级住宅区，这样看来，中大的投资前景很大啊！”\n不过，...   
77  579edaad  明星们的经纪人助理很多时候都是跟明星形影不离的存在，所以说很多时候最了解明星的可能就是经纪人...   
78  cdf7d348  今年这个年代，大家对贷款这个字眼肯定都是不陌生的吧。毕竟社会在发展，无论是大家平时的衣食住行...   
79  b4bb3ad0  尚智逢源注意到，多只基金发行启用明星基金经理，欲在今年打造公司的权益基金明星产品。2月开始，...   
80  5b655e03  \n开餐厅干得不好，就像是生病了没治好，不死不活，拖累别人自己也不舒服。餐厅一旦出现了半死不...   
81  39bf0855  税务行政处罚是纳税人违反税务行政管理秩序而依法应当承担的税务行政法律责任，纳税人对税务行政处...   
82  e5ecec51  不少的人在买车之后会发出这样的感叹，买车容易养车难，现在的车几万就能买一辆，为什么说用车难，...   
83  5bd7cc02  在股市里，庄家有很多炒作手法，虽然目的不尽相同，但是最终都是为了迷惑散户，达到自己赚钱的目的...   
84  0f570860  各位亲！\n增值税小规模纳税人申报流程\n有变化哦！\n复杂么？\nno！no！no！\n超...   
85  4beccbf6  3年前，刚刚工作，只有数千元的工资在一线大城市让我感到非常的不适，交房租，吃饭，一个月的工资...   
86  a4199b45  如何保障可燃气体报警器的生产安全?可燃气体报警仪是关系到安全的，所以属于强制检定的范围。可燃...   
87  b0997221  近日，0glass宣布已经完成数千万b轮融资，本轮投资由清研新一代人工智能基金（珠海、嘉善）...   
88  9268995e  \n企业站\n上一文我们说了应用比较广泛的几大类型的网站，今天我们细说一下企业站。\n企业站...   
89  5f5ca51c  交易理念是投资交易的理论基础，有扎实的交易理念，有合理的交易计划。期货市场是以现货市场为依托...   
90  ca05670e  2019年3月1日，北京创景咨询有限公司从众多竞标公司中脱颖而出，成功中标复旦大学管理学院教...   
91  9cec002e  dc已公布2019年及以后的手机市场预测，预计会出现更多困难时期。今年将是全球智能手机出货量...   
92  31820274  \n（免责声明和郑重提醒：本文内容纯属作者个人观点，仅供投资者参考，不构成投资建议。）\n\...   
93  05491ec8  引流有15种类型：今日主要来分享一下：价格引流、低价、真实2.员工引流、发动员工身边的人脉3...   
94  d4cbe167  \newn分析师：btc迈入上破关键区域，4000关口成多方发力重要踏板\n\n\nethe...   
95  08461740  今天的大盘贼恐怖，尤其是尾盘一小时，先是暴跌，然后大涨，最后收盘突破3100点，还创出了近3...   
96  457829ab  有个懂你的人是世上最大的幸福，最大的幸运！有个懂你的人，心里最温暖；有个懂你的人，心里最甜蜜...   
97  935c3546  一、公司业务概况及财务状况分析\n（一）公司业务概况\n陕西中远机械有限责任公

In [143]:
df1['top_esm_entity']=df1.esm_content.map(lambda x:post(x,topk=30,frc=2))

In [144]:
df1['top_esm_title']=df1.esm_title.map(lambda x:post(x,topk=5,frc=0))

In [145]:
df1.head(30)

newsId                                            content  \
0   4e36d02a  第十七届上海国际工业自动化及机器人展与上海智能工厂展览会于2019年3月1日圆满落下帷幕。展...   
1   cb8e8b79  大数据之热度，已无需多言。业内众多关于大数据可视化应用领域的声音与讨论，大多集中在行业领域和...   
2   f85c18e2  过年期间和一些搞运输的老朋友聊天，大家始终绕不开的一个话题：“如今物流行业越来越难干了！”想...   
3   889199b4  水泥路面起砂如何处理？修补后材料会发生脱落吗？水泥路面起砂病害的发生可能是材料老化的现象，也...   
4   a811d20e  我们今天再开始复盘之前，先简单说下行情，昨天大幅度波动90点的时候，很多人看空，但是却没有勇...   
5   4e798999  对于大部分入门玩家来说，在电源的选择上往往都比较随意。他们在意的方面一般只有功率的大小，而对...   
6   1f5679fb  新打的水泥地面受冻掉皮怎么修复？水泥路面、地面掉皮病害发生后，有一些水泥路面、地面发生病害后...   
7   1a8339ea  一、信托融资概念及特点\n信托是以资产为核心，信任为基础，委托为方式的财产管理制度。信托融资...   
8   51203867  当“租购并举”成为国策，住房租赁的庞大市场需求和政策红利的持续加持，让越来越多人发现并逐步重...   
9   5bb9047a  随着生活压力和工作压力都不断增加的情况下，有心理问题的人也越来越多，很多人都开始求助心理咨询...   
10  478af383  标普500成分股公司在去年第四季度关心什么？摩根大通从它们的电话会议记录、问答环节记录以及成...   
11  f0a7d8d9  随着移动支付的便民化与普及，现在基本上在三线城市以上，不论走到何地，手机一扫，分分钟就能付款...   
12  afee8716  新倚天屠龙记：灭绝硬撑最后一口气，爆出惊人秘密，让她崩溃大哭\n大型古装武侠剧《新倚天屠龙记...   
13  efd34fe8  中字头国企整体产权转让。公司名下核心资产为廊坊市区三块工业用地，合计面积约220亩。公司自有...   
14  73ce42e1  玩具行业一周简讯，宜采网整理报道：\n【简讯一】\n喜讯！制造业税率降3个点，电价降10%！...   
15  783adc02  江苏天鼎看股票：盘中盯花眼，不如盘前下功夫！\n\n在前文中江苏天鼎简单为大家介绍了集合竞价...   
16  960b8e04  2018年债牛行情一骑绝尘，2019年a股“牛市”横空演绎，债市耀眼的光环就这样被a股取代了...   
17  d7cb8b85  \n为什么下跌的黄金走到1282至1280一线之后，却停止不前了，到底是见底了还是在酝酿反弹...   
18  66fb9c0c  导读：蔚来汽车三年累计亏损已达172.3亿。不过，从积极的一面来看，随着蔚来的第二款量产车型...   
19  cd510c0d  \n胡润研究院携手小小世界全球魅力女性商学院3月7日联合发布《2019胡润全球白手起家女富豪...   
20  80800a17  今天市场不敢发表太多，但是大家留意两会后，还有近期会有大幅度的波动，好似蹦迪那样的感觉。如果...   
21  2322249a  实验室建设不同于普通的建设项目，具有综合性、专业性、系统性强的特点，必须统一部署，总体协调，...   
22  ac148da5  印度标准局（bis）是印度标准及认证事务的主管部门。\n\n历史沿革\n\n印度在独立前没有...   
23  fe0ae621  上篇文章我们来了解了亚马逊跨境电商的优点，便捷链接：花无百日红，唯你一支独秀，亚马逊跨境电商...   
24  530ccf21  商场触摸查询广告机是交互3d高科技以及互联网科技来和谐商业连锁店与顾客间在出售、营销和服务上...   
25  7885555d  ​​百度竞价推广是一种按效果收费的广告，是目前互联网行业较为精准的广告投放方法之一，而随着愈...   
26  c494a2ab  虽然科技发展，社会进步，但很多人特别是老一辈人在子女的婚恋方面还是多数会选择相亲的方式。而与...   
27  779320d5  2019年1月25日，重庆水利电力职业技术学院水利工程学院副院长闵志华，水工造价专任教师雷丽...   
28  46f7801f  三月阳光正好，三月微风不燥，\n春暖花开，柳暖花香，正如你如此美好\n确认过眼神，遇见最美丽...   
29  ed3a5d29  \n作为企业管理者，担当是必备的素质之一，你懂得如何担当吗？而“担当”的内涵包括：提升能力，...   

                                title                         entity  \
0                   sia智慧工厂展，誉洋以“智”取胜                     3d 工业 机器视觉   
1                        大数据可视化应用领域探析                      数据 卫星 可视化   
2             大车发大财的时代一去不返？货运行业还有春天吗？                       物流 货运 信息   
3            水泥路面起砂如何处理？修补后材料会发生脱落吗？                        路面 水泥 病害   
4             股市高位还将继续涨， 这几类股票有望成为新热点                       阳线 股票 涨势   
5                 装机容易忽视的组件，主机电源不能随便选                     evga 电源 组件   
6           新打的水泥地面受冻掉皮怎么修复？修补质量如何保证？                       水泥 病害 路面   
7               信托融资模式下房地产信托融资及发展趋势解析                      信托 房地产 股权   
8                 地产基金投资公寓资产，抓好这两点是关键                   租购并举 公寓资产 租客   
9                       心理咨询小程序开发案例分析                 心理 心理咨询师 小程序开发   
10   摩根大通分析了25000份公司成绩单发现 对地缘政治的担忧在加剧                标普500 摩根大通 地缘政治   
11           商家只支持移动支付？央行表示：拒收现金涉嫌违法！               移动支付 人民币 中华人民共和国   
12     新倚天屠龙记：灭绝硬撑最后一口气，爆出惊人秘密，让她崩溃大哭                     周芷若 杨逍 纪晓芙   
13                       廊坊220亩工业用地转让                    廊坊市 建筑面积 土地   
14     制造业税率降3个点，电价降10%！多起火灾！“光头强”侵权案                      宜采网 简讯 工厂   
15             江苏天鼎看股票：盘中盯花眼，不如盘前下功夫！                   集合竞价 大盘 江苏天鼎   
16                  2019年，股债双牛的概率有多大？                    股债双牛 债券 薛掌柜   
17          策略家张伟：金银低位震荡勿追空，反弹走势将会来临！                       一线 价格 走势   
18  去年净亏96亿！马化腾、刘强东投资的“中国特斯拉”3年亏掉172亿                   特斯拉 人民币 商业王道   
19      胡润全球白手起家女富豪榜：57%全球最成功女企业家来自中国     2019胡润全球白手起家女富豪榜 胡润研究院 吴亚军   
20                  【3月7日跟踪】低价+空间票！！！                     短线 京威股份 低价   
21                      实验室建设规划方案怎么写？                      实验室 方案 领导   
22                          印度bis认证介绍                   印度标准局 印度 bis   
23        亚马逊跨境电商解答：什么是无货源模式，什么是大量铺货？                      亚马逊 淘宝 店铺   
24             节假日使用商场触摸查询广告机能够带来什么效果                     广告机 商场 消费者   
25        竞价广告并非完全是有钱人的游戏，你需要了解的是投放策略                   

In [146]:
df1.apply(lambda x:len(set(x['top_esm_entity']+x['top_esm_title'])),axis=1).sum()

718038

In [147]:
df1['top_in_cnt'] = df1.apply(lambda x:len(set(x['top_esm_entity']+x['esm_title'].split()) & set(x['entity'].split())),axis=1)

In [148]:
df1['entity_cnt'].sum()

87337

In [149]:
df1['entity_cnt'] = df1['entity'].apply(lambda x:len(x.split()))

In [150]:
float(df1.top_in_cnt.sum())/float(df1.entity_cnt.sum())

0.7580177931460893

#

词频>0:0.86    
词频>1:0.8022945601520547    
    0.76   1277422
    
top 30:    0.75   922546


content 
top =30 frc=1:    0.782   993440    
top =10 frc=1:    0.729   487022    

top =10 frc=1 top =5 frc=0:    0.729   423132      
top =10 frc=2 top =5 frc=0:    0.724   400176      
 top =30 frc=2 top =5 frc=0:    0.758                   718038                

In [151]:
df1.head()

newsId                                            content  \
0  4e36d02a  第十七届上海国际工业自动化及机器人展与上海智能工厂展览会于2019年3月1日圆满落下帷幕。展...   
1  cb8e8b79  大数据之热度，已无需多言。业内众多关于大数据可视化应用领域的声音与讨论，大多集中在行业领域和...   
2  f85c18e2  过年期间和一些搞运输的老朋友聊天，大家始终绕不开的一个话题：“如今物流行业越来越难干了！”想...   
3  889199b4  水泥路面起砂如何处理？修补后材料会发生脱落吗？水泥路面起砂病害的发生可能是材料老化的现象，也...   
4  a811d20e  我们今天再开始复盘之前，先简单说下行情，昨天大幅度波动90点的时候，很多人看空，但是却没有勇...   

                      title      entity         emotion  \
0         sia智慧工厂展，誉洋以“智”取胜  3d 工业 机器视觉     POS POS POS   
1              大数据可视化应用领域探析   数据 卫星 可视化     POS POS POS   
2   大车发大财的时代一去不返？货运行业还有春天吗？    物流 货运 信息     POS NEG POS   
3  水泥路面起砂如何处理？修补后材料会发生脱落吗？     路面 水泥 病害  NORM NORM NORM   
4   股市高位还将继续涨， 这几类股票有望成为新热点    阳线 股票 涨势     POS POS POS   

                                           esm_title  \
0                sia   智慧   智慧工厂                       
1                         大数 大数据         应用   领域       
2    大车           时代             货运   行业 还     春天...   
3    水泥   路面                       材料                  
4  股 股市 高 高位 还   继              类股 股票             热点   

                                         esm_content  \
0            上海 上海国 国际   工业     工业自动化     机器 机器人 ...   
1    大数 大数据     热度                 业内 内众         ...   
2      年期 期间                   老朋友   聊天     大家   ...   
3    水泥   路面                       材料            ...   
4                  复盘                 下行 行情      ...   

                                      top_esm_entity  top_in_cnt  entity_cnt  \
0  [机器, 智能, 工业, 企业, 机器视觉, 3d, 机器人, 展会, 工业自动化, 技术,...           3           3   
1  [智能, 数据, 领域, 应用, 大数, 大数据, 决策, 卫星, 态势, 商业, 设备, ...           2           3   
2  [物流, 货运, 信息, 行业, 服务, 顺风, 货源, 货主, 同城, 大车, 道路, 资...           3           3   
3      [路面, 水泥, 水平, 材料, 病害, 原因, 可能, 现象, 问题, 表层, 粘结力]           3           3   
4       [上涨, 就是, 市场, 趋势, 行业, 股票, 阳线, 前期, 个股, 消费, 涨势]           3           3   

       top_esm_title  
0    [sia, 智慧, 智慧工厂]  
1  [大数, 大数据, 应用, 领域]  
2   [大车, 时代, 货运, 行业]  
3       [水泥, 路面, 材料]  
4           [股市, 高位]

In [156]:


newlines=[]
for i,line in df1.head().iterrows():


    texta =line['content']
    entity_real = line['entity'].split()
    entity_recall = line['top_esm_title'] + line['top_esm_entity']
    for entity in entity_recall:
        textb = entity
        if entity in entity_real:
            label = 1
        else:
            label = 0
        newlines.append([line['newsId'],texta,textb,label])
df_pair = pd.DataFrame(newlines,columns=['newsId','texta','textb','label'])

In [181]:
gg = df_pair[df_pair.label==1][['newsId','textb']].groupby(['newsId'])
df_agg2 = gg['textb'].apply(lambda x:','.join(x)).reset_index()
df_agg2['textb']=df_agg2['textb'].map(lambda x:[i[0] for i in Counter(x.split()).most_common(3)])
df_agg2['textb']=df_agg2['textb'].apply(lambda x:','.join(x))


In [183]:
df_pair.head()

newsId                                              texta textb  label
0  4e36d02a  第十七届上海国际工业自动化及机器人展与上海智能工厂展览会于2019年3月1日圆满落下帷幕。展...   sia      0
1  4e36d02a  第十七届上海国际工业自动化及机器人展与上海智能工厂展览会于2019年3月1日圆满落下帷幕。展...    智慧      0
2  4e36d02a  第十七届上海国际工业自动化及机器人展与上海智能工厂展览会于2019年3月1日圆满落下帷幕。展...  智慧工厂      0
3  4e36d02a  第十七届上海国际工业自动化及机器人展与上海智能工厂展览会于2019年3月1日圆满落下帷幕。展...    机器      0
4  4e36d02a  第十七届上海国际工业自动化及机器人展与上海智能工厂展览会于2019年3月1日圆满落下帷幕。展...    智能      0